<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Belinda Phipson" />


<title>Analysis of adult samples</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Porello-heart-snRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://git.mcri.edu.au/bioinformatics/Porello-heart-snRNAseq">
    <span class="fa fa-gitlab"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Analysis of adult samples</h1>
<h4 class="author">Belinda Phipson</h4>
<h4 class="date">6/21/2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-07-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Porello-heart-snRNAseq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.3.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190603code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190603)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190603code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190603)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgitmcrieduaubioinformaticsporelloheartsnrnaseqtreeda40aba9c2abd915330f0e1316870506a88cab87targetblankda40abaa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/tree/da40aba9c2abd915330f0e1316870506a88cab87" target="_blank">da40aba</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgitmcrieduaubioinformaticsporelloheartsnrnaseqtreeda40aba9c2abd915330f0e1316870506a88cab87targetblankda40abaa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/05-ClusterDCM.Rmd
    Untracked:  code/ReadDataObjects.R
    Untracked:  data/gstlist-fetal.Rdata
    Untracked:  data/gstlist-young.Rdata
    Untracked:  data/heart-markers-long.txt
    Untracked:  data/pseudobulk.Rds
    Untracked:  output/AllAdult-clustermarkers.csv
    Untracked:  output/AllFetal-clustermarkers.csv
    Untracked:  output/AllYoung-clustermarkers.csv
    Untracked:  output/Alldcm-clustermarkers.csv
    Untracked:  output/Figures/
    Untracked:  output/MarkerAnalysis/
    Untracked:  output/RDataObjects/
    Untracked:  output/fetal1-clustermarkers.csv
    Untracked:  output/fetal2-clustermarkers.csv
    Untracked:  output/fetal3-clustermarkers.csv
    Untracked:  output/heatmap-top10-adultmarkergenes.pdf
    Untracked:  output/young1-clustermarkers.csv

Unstaged changes:
    Modified:   analysis/01-QualityControl.Rmd
    Modified:   analysis/01a-DEpseudobulk.Rmd
    Modified:   analysis/02-ClusterFetal.Rmd
    Modified:   analysis/02c-ClusterFetal3.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/da40aba9c2abd915330f0e1316870506a88cab87/analysis/04-ClusterAdult.Rmd" target="_blank">da40aba</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
<td>
update young and adult analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/04-ClusterAdult.html" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/6fb2f736464637cfaaffbb26bbdbb2e102b8eb6e/analysis/04-ClusterAdult.Rmd" target="_blank">6fb2f73</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
<td>
add adult analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>I will cluster all the adult samples from batch 1 and batch 2 using the integration technique from the Seurat package.</p>
</div>
<div id="load-libraries-and-functions" class="section level1">
<h1>Load libraries and functions</h1>
<pre class="r"><code>library(edgeR)</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre class="r"><code>library(RColorBrewer)
library(org.Hs.eg.db)</code></pre>
<pre><code>Loading required package: AnnotationDbi</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>Loading required package: parallel</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:parallel&#39;:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>The following object is masked from &#39;package:limma&#39;:

    plotMA</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter,
    Find, get, grep, grepl, intersect, is.unsorted, lapply, Map,
    mapply, match, mget, order, paste, pmax, pmax.int, pmin,
    pmin.int, Position, rank, rbind, Reduce, rownames, sapply,
    setdiff, sort, table, tapply, union, unique, unsplit, which,
    which.max, which.min</code></pre>
<pre><code>Loading required package: Biobase</code></pre>
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    expand.grid</code></pre>
<pre><code></code></pre>
<pre class="r"><code>library(limma)
library(Seurat)</code></pre>
<pre><code>Registered S3 method overwritten by &#39;R.oo&#39;:
  method        from       
  throw.default R.methodsS3</code></pre>
<pre><code>Registered S3 methods overwritten by &#39;ggplot2&#39;:
  method         from 
  [.quosures     rlang
  c.quosures     rlang
  print.quosures rlang</code></pre>
<pre class="r"><code>library(monocle)</code></pre>
<pre><code>Loading required package: Matrix</code></pre>
<pre><code>
Attaching package: &#39;Matrix&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:S4Vectors&#39;:

    expand</code></pre>
<pre><code>Loading required package: ggplot2</code></pre>
<pre><code>Loading required package: VGAM</code></pre>
<pre><code>Loading required package: splines</code></pre>
<pre><code>Loading required package: DDRTree</code></pre>
<pre><code>Loading required package: irlba</code></pre>
<pre class="r"><code>library(cowplot)</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggplot2&#39;:

    ggsave</code></pre>
<pre class="r"><code>library(DelayedArray)</code></pre>
<pre><code>Loading required package: matrixStats</code></pre>
<pre><code>
Attaching package: &#39;matrixStats&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:Biobase&#39;:

    anyMissing, rowMedians</code></pre>
<pre><code>Loading required package: BiocParallel</code></pre>
<pre><code>
Attaching package: &#39;DelayedArray&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    aperm, apply, rowsum</code></pre>
<pre class="r"><code>library(scran)</code></pre>
<pre><code>Loading required package: SingleCellExperiment</code></pre>
<pre><code>Loading required package: SummarizedExperiment</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>
Attaching package: &#39;SingleCellExperiment&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:edgeR&#39;:

    cpm</code></pre>
<pre class="r"><code>library(NMF)</code></pre>
<pre><code>Loading required package: pkgmaker</code></pre>
<pre><code>Loading required package: registry</code></pre>
<pre><code>
Attaching package: &#39;pkgmaker&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:S4Vectors&#39;:

    new2</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    isFALSE</code></pre>
<pre><code>Loading required package: rngtools</code></pre>
<pre><code>Loading required package: cluster</code></pre>
<pre><code>NMF - BioConductor layer [OK] | Shared memory capabilities [NO: synchronicity] | Cores 31/32</code></pre>
<pre><code>  To enable shared memory capabilities, try: install.extras(&#39;
NMF
&#39;)</code></pre>
<pre><code>
Attaching package: &#39;NMF&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:DelayedArray&#39;:

    seed</code></pre>
<pre><code>The following object is masked from &#39;package:BiocParallel&#39;:

    register</code></pre>
<pre><code>The following object is masked from &#39;package:S4Vectors&#39;:

    nrun</code></pre>
<pre class="r"><code>library(workflowr)</code></pre>
<pre><code>This is workflowr version 1.3.0
Run ?workflowr for help getting started</code></pre>
<pre class="r"><code>library(ggplot2)
library(clustree)</code></pre>
<pre><code>Loading required package: ggraph</code></pre>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:GenomicRanges&#39;:

    intersect, setdiff, union</code></pre>
<pre><code>The following object is masked from &#39;package:GenomeInfoDb&#39;:

    intersect</code></pre>
<pre><code>The following object is masked from &#39;package:matrixStats&#39;:

    count</code></pre>
<pre><code>The following object is masked from &#39;package:AnnotationDbi&#39;:

    select</code></pre>
<pre><code>The following objects are masked from &#39;package:IRanges&#39;:

    collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>The following objects are masked from &#39;package:S4Vectors&#39;:

    first, intersect, rename, setdiff, setequal, union</code></pre>
<pre><code>The following object is masked from &#39;package:Biobase&#39;:

    combine</code></pre>
<pre><code>The following objects are masked from &#39;package:BiocGenerics&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>source(&quot;/misc/card2-single_cell_nuclei_rnaseq/Porello-heart-snRNAseq/code/normCounts.R&quot;)
source(&quot;/misc/card2-single_cell_nuclei_rnaseq/Porello-heart-snRNAseq/code/findModes.R&quot;)
source(&quot;/misc/card2-single_cell_nuclei_rnaseq/Porello-heart-snRNAseq/code/ggplotColors.R&quot;)</code></pre>
</div>
<div id="read-in-the-adult-data" class="section level1">
<h1>Read in the adult data</h1>
<pre class="r"><code>targets &lt;- read.delim(&quot;/misc/card2-single_cell_nuclei_rnaseq/Porello-heart-snRNAseq/data/targets.txt&quot;,header=TRUE, stringsAsFactors = FALSE)
targets$FileName2 &lt;- paste(targets$FileName,&quot;/&quot;,sep=&quot;&quot;)
targets$Group_ID2 &lt;- gsub(&quot;LV_&quot;,&quot;&quot;,targets$Group_ID)
group &lt;- c(&quot;Fetal_1&quot;,&quot;Fetal_2&quot;,&quot;Fetal_3&quot;,
           &quot;Young_1&quot;,&quot;Young_2&quot;,&quot;Young_3&quot;,
           &quot;Adult_1&quot;,&quot;Adult_2&quot;,&quot;Adult_3&quot;, 
           &quot;Diseased_1&quot;,&quot;Diseased_2&quot;,
           &quot;Diseased_3&quot;,&quot;Diseased_4&quot;)
m &lt;- match(group, targets$Group_ID2)
targets &lt;- targets[m,]

a1 &lt;- Read10X(data.dir = targets$FileName2[targets$Group_ID2==&quot;Adult_1&quot;])
colnames(a1) &lt;- paste(colnames(a1),&quot;a1&quot;,sep=&quot;_&quot;)
a2 &lt;- Read10X(data.dir = targets$FileName2[targets$Group_ID2==&quot;Adult_2&quot;])
colnames(a2) &lt;- paste(colnames(a2),&quot;a2&quot;,sep=&quot;_&quot;)
a3 &lt;- Read10X(data.dir = targets$FileName2[targets$Group_ID2==&quot;Adult_3&quot;])
colnames(a3) &lt;- paste(colnames(a3),&quot;a3&quot;,sep=&quot;_&quot;)

# Combine 3 samples into one big data matrix
alla &lt;- cbind(a1,a2,a3)</code></pre>
</div>
<div id="gene-filtering" class="section level1">
<h1>Gene filtering</h1>
<div id="get-gene-annotation" class="section level2">
<h2>Get gene annotation</h2>
<p>I’m using gene annotation information from the <code>org.Hs.eg.db</code> package.</p>
<pre class="r"><code>columns(org.Hs.eg.db)</code></pre>
<pre><code> [1] &quot;ACCNUM&quot;       &quot;ALIAS&quot;        &quot;ENSEMBL&quot;      &quot;ENSEMBLPROT&quot; 
 [5] &quot;ENSEMBLTRANS&quot; &quot;ENTREZID&quot;     &quot;ENZYME&quot;       &quot;EVIDENCE&quot;    
 [9] &quot;EVIDENCEALL&quot;  &quot;GENENAME&quot;     &quot;GO&quot;           &quot;GOALL&quot;       
[13] &quot;IPI&quot;          &quot;MAP&quot;          &quot;OMIM&quot;         &quot;ONTOLOGY&quot;    
[17] &quot;ONTOLOGYALL&quot;  &quot;PATH&quot;         &quot;PFAM&quot;         &quot;PMID&quot;        
[21] &quot;PROSITE&quot;      &quot;REFSEQ&quot;       &quot;SYMBOL&quot;       &quot;UCSCKG&quot;      
[25] &quot;UNIGENE&quot;      &quot;UNIPROT&quot;     </code></pre>
<pre class="r"><code>ann &lt;- AnnotationDbi:::select(org.Hs.eg.db,keys=rownames(alla),columns=c(&quot;SYMBOL&quot;,&quot;ENTREZID&quot;,&quot;ENSEMBL&quot;,&quot;GENENAME&quot;,&quot;CHR&quot;),keytype = &quot;SYMBOL&quot;)</code></pre>
<pre><code>&#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<pre class="r"><code>m &lt;- match(rownames(alla),ann$SYMBOL)
ann &lt;- ann[m,]
table(ann$SYMBOL==rownames(alla))</code></pre>
<pre><code>
 TRUE 
33939 </code></pre>
<pre class="r"><code>mito &lt;- grep(&quot;mitochondrial&quot;,ann$GENENAME)
length(mito)</code></pre>
<pre><code>[1] 226</code></pre>
<pre class="r"><code>ribo &lt;- grep(&quot;ribosomal&quot;,ann$GENENAME)
length(ribo)</code></pre>
<pre><code>[1] 198</code></pre>
<pre class="r"><code>missingEZID &lt;- which(is.na(ann$ENTREZID))
length(missingEZID)</code></pre>
<pre><code>[1] 10530</code></pre>
</div>
<div id="remove-mitochondrial-and-ribosomal-genes-and-genes-with-no-entrezid" class="section level2">
<h2>Remove mitochondrial and ribosomal genes and genes with no ENTREZID</h2>
<p>These genes are not informative for downstream analysis.</p>
<pre class="r"><code>chuck &lt;- unique(c(mito,ribo,missingEZID))
length(chuck)</code></pre>
<pre><code>[1] 10875</code></pre>
<pre class="r"><code>alla.keep &lt;- alla[-chuck,]
ann.keep &lt;- ann[-chuck,]
table(ann.keep$SYMBOL==rownames(alla.keep))</code></pre>
<pre><code>
 TRUE 
23064 </code></pre>
</div>
<div id="remove-very-lowly-expressed-genes" class="section level2">
<h2>Remove very lowly expressed genes</h2>
<p>Removing very lowly expressed genes helps to reduce the noise in the data. Here I am choosing to keep genes with at least 1 count in at least 20 cells. This means that a cluster made up of at least 20 cells can potentially be detected (minimum cluster size = 20 cells).</p>
<pre class="r"><code>numzero.genes &lt;- rowSums(alla.keep==0)

#avg.exp &lt;- rowMeans(cpm.DGEList(y.kid,log=TRUE))

#plot(avg.exp,numzero.genes,xlab=&quot;Average log-normalised-counts&quot;,ylab=&quot;Number zeroes per gene&quot;)

table(numzero.genes &gt; (ncol(alla.keep)-20))</code></pre>
<pre><code>
FALSE  TRUE 
17605  5459 </code></pre>
<pre class="r"><code>keep.genes &lt;- numzero.genes &lt; (ncol(alla.keep)-20)
table(keep.genes)</code></pre>
<pre><code>keep.genes
FALSE  TRUE 
 5518 17546 </code></pre>
<pre class="r"><code>alla.keep &lt;- alla.keep[keep.genes,]
dim(alla.keep)</code></pre>
<pre><code>[1] 17546  9416</code></pre>
<pre class="r"><code>ann.keep &lt;- ann.keep[keep.genes,]</code></pre>
<p>The total size of the adult dataset is 9416 cells and 17546 genes.</p>
</div>
</div>
<div id="remove-sex-chromosome-genes" class="section level1">
<h1>Remove sex chromosome genes</h1>
<p>I will remove the sex chromosome genes before clustering so that the sex doesn’t play a role in determining the clusters.</p>
<pre class="r"><code>sexchr &lt;- ann.keep$CHR %in% c(&quot;X&quot;,&quot;Y&quot;)

alla.nosex &lt;- alla.keep[!sexchr,]
dim(alla.nosex)</code></pre>
<pre><code>[1] 16900  9416</code></pre>
<pre class="r"><code>ann.nosex &lt;- ann.keep[!sexchr,]</code></pre>
</div>
<div id="save-data-objects" class="section level1">
<h1>Save data objects</h1>
<pre class="r"><code>save(ann,ann.keep,ann.nosex,alla,alla.keep,alla.nosex,file=&quot;./output/RDataObjects/adultObjs.Rdata&quot;)</code></pre>
</div>
<div id="create-seurat-objects" class="section level1">
<h1>Create Seurat objects</h1>
<p>Here I am following the Seurat vignette on performing clustering using their integration method to deal with batch effects.</p>
<pre class="r"><code>biorep &lt;- factor(rep(c(&quot;a1&quot;,&quot;a2&quot;,&quot;a3&quot;),c(ncol(a1),ncol(a2),ncol(a3))))
names(biorep) &lt;- colnames(alla.keep)
sex &lt;- factor(rep(c(&quot;f&quot;,&quot;m&quot;,&quot;m&quot;),c(ncol(a1),ncol(a2),ncol(a3))))
names(sex) &lt;- colnames(alla.keep)
age &lt;- rep(c(35,41,42),c(ncol(a1),ncol(a2),ncol(a3)))
names(age) &lt;- colnames(alla.keep)
batch &lt;- rep(c(&quot;B2&quot;,&quot;B2&quot;,&quot;B1&quot;),c(ncol(a1),ncol(a2),ncol(a3)))
names(batch) &lt;- colnames(alla.keep)

adult &lt;- CreateSeuratObject(counts = alla.nosex, project = &quot;adult&quot;)
adult &lt;- AddMetaData(object=adult, metadata = biorep, col.name=&quot;biorep&quot;)
adult &lt;- AddMetaData(object=adult, metadata = sex, col.name=&quot;sex&quot;)
adult &lt;- AddMetaData(object=adult, metadata = age, col.name=&quot;age&quot;)
adult &lt;- AddMetaData(object=adult, metadata = batch, col.name=&quot;batch&quot;)</code></pre>
<pre class="r"><code>adult.list &lt;- SplitObject(adult, split.by = &quot;biorep&quot;)</code></pre>
</div>
<div id="try-new-normalisation-method-sctransform" class="section level1">
<h1>Try new normalisation method: SCTransform</h1>
<p>This new method replaces the <code>NormalizeData</code>, <code>FindVariableFeatures</code> and <code>ScaleData</code> functions. It performs regularised negative binomial regression with the total sequencing depth per cell as the covariate (i.e. library size), as well as any other user supplied covariates. The Pearson residuals are then used in downstream analysis.</p>
<pre class="r"><code># This is a bit slow
for (i in 1:length(adult.list)) {
    adult.list[[i]] &lt;- SCTransform(adult.list[[i]], verbose = FALSE)
#    adult.list[[i]] &lt;- GetResidual(adult.list[[i]])
}</code></pre>
</div>
<div id="perform-the-usual-normalisation" class="section level1">
<h1>Perform the usual normalisation</h1>
<pre class="r"><code>#for (i in 1:length(adult.list)) {
#    adult.list[[i]] &lt;- NormalizeData(adult.list[[i]], verbose = FALSE)
#    adult.list[[i]] &lt;- FindVariableFeatures(adult.list[[i]], selection.method #= &quot;vst&quot;, 
#                                            nfeatures = 2000, verbose = FALSE)
#}</code></pre>
</div>
<div id="perform-integration" class="section level1">
<h1>Perform integration</h1>
<p>There are two steps:</p>
<ul>
<li>Find integration anchors</li>
<li>Perform integration which should batch-correct the data</li>
</ul>
<p>The default number of dimensions is 30. Should increase the number of integration anchors to 3000.</p>
<pre class="r"><code>adult.anchors &lt;- FindIntegrationAnchors(object.list = adult.list, dims = 1:30, anchor.features = 3000)</code></pre>
<pre><code>Computing 3000 integration features</code></pre>
<pre><code>Scaling features for provided objects</code></pre>
<pre><code>Finding all pairwise anchors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 7668 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 5349 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 5109 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 4374 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre><code>Running CCA</code></pre>
<pre><code>Merging objects</code></pre>
<pre><code>Finding neighborhoods</code></pre>
<pre><code>Finding anchors</code></pre>
<pre><code>    Found 4782 anchors</code></pre>
<pre><code>Filtering anchors</code></pre>
<pre><code>    Retained 4071 anchors</code></pre>
<pre><code>Extracting within-dataset neighbors</code></pre>
<pre class="r"><code>adult.integrated &lt;- IntegrateData(anchorset = adult.anchors, dims = 1:30)</code></pre>
<pre><code>Merging dataset 3 into 1</code></pre>
<pre><code>Extracting anchors for merged samples</code></pre>
<pre><code>Finding integration vectors</code></pre>
<pre><code>Finding integration vector weights</code></pre>
<pre><code>Integrating data</code></pre>
<pre><code>Merging dataset 2 into 1 3</code></pre>
<pre><code>Extracting anchors for merged samples</code></pre>
<pre><code>Finding integration vectors</code></pre>
<pre><code>Finding integration vector weights</code></pre>
<pre><code>Integrating data</code></pre>
</div>
<div id="perform-clustering" class="section level1">
<h1>Perform clustering</h1>
<pre class="r"><code>DefaultAssay(object = adult.integrated) &lt;- &quot;integrated&quot;</code></pre>
<div id="perform-scaling-and-pca" class="section level2">
<h2>Perform scaling and PCA</h2>
<pre class="r"><code>adult.integrated &lt;- ScaleData(adult.integrated, verbose = FALSE)
adult.integrated &lt;- RunPCA(adult.integrated, npcs = 50, verbose = FALSE)
ElbowPlot(adult.integrated,ndims=50)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/pca-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-1">
Past versions of pca-1.png
</button>
</p>
<div id="fig-pca-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/pca-1.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VizDimLoadings(adult.integrated, dims = 1:4, reduction = &quot;pca&quot;)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-14-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-14-1.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(adult.integrated, reduction = &quot;pca&quot;,group.by=&quot;biorep&quot;)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-15-1.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(adult.integrated, reduction = &quot;pca&quot;,group.by=&quot;sex&quot;)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-2">
Past versions of unnamed-chunk-15-2.png
</button>
</p>
<div id="fig-unnamed-chunk-15-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-15-2.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(adult.integrated, reduction = &quot;pca&quot;,group.by=&quot;batch&quot;)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-15-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-3">
Past versions of unnamed-chunk-15-3.png
</button>
</p>
<div id="fig-unnamed-chunk-15-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-15-3.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimHeatmap(adult.integrated, dims = 1:15, cells = 500, balanced = TRUE)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-16-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-16-1.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimHeatmap(adult.integrated, dims = 16:30, cells = 500, balanced = TRUE)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-16-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-2">
Past versions of unnamed-chunk-16-2.png
</button>
</p>
<div id="fig-unnamed-chunk-16-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-16-2.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="perform-nearest-neighbours-clustering" class="section level2">
<h2>Perform nearest neighbours clustering</h2>
<pre class="r"><code>adult.integrated &lt;- FindNeighbors(adult.integrated, dims = 1:20)</code></pre>
<pre><code>Computing nearest neighbor graph</code></pre>
<pre><code>Computing SNN</code></pre>
<pre class="r"><code>adult.integrated &lt;- FindClusters(adult.integrated, resolution = 0.6)</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9416
Number of edges: 354612

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9127
Number of communities: 21
Elapsed time: 0 seconds</code></pre>
<pre class="r"><code>table(Idents(adult.integrated))</code></pre>
<pre><code>
   0    1    2    3    4    5    6    7    8    9   10   11   12   13   14 
1845 1074  928  724  669  650  546  516  398  389  330  212  210  199  133 
  15   16   17   18   19   20 
 121  117  114   92   84   65 </code></pre>
</div>
<div id="visualisation-with-tsne" class="section level2">
<h2>Visualisation with TSNE</h2>
<pre class="r"><code>set.seed(10)
adult.integrated &lt;- RunTSNE(adult.integrated, reduction = &quot;pca&quot;, dims = 1:20)</code></pre>
<pre class="r"><code>pdf(file=&quot;./output/Figures/tsne-adultALL.pdf&quot;,width=10,height=8,onefile = FALSE)
DimPlot(adult.integrated, reduction = &quot;tsne&quot;,label=TRUE,label.size = 6,pt.size = 0.5)+NoLegend()
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>DimPlot(adult.integrated, reduction = &quot;tsne&quot;,label=TRUE,label.size = 6)+NoLegend()</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-18-1.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(adult.integrated, reduction = &quot;tsne&quot;, split.by = &quot;biorep&quot;,label=TRUE,label.size = 5)+NoLegend()</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-19-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-19-1.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(adult.integrated, reduction = &quot;tsne&quot;, split.by = &quot;sex&quot;,label=TRUE,label.size = 5)+NoLegend()</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-20-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-20-1.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(adult.integrated, reduction = &quot;tsne&quot;, group.by = &quot;biorep&quot;)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-21-1.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(adult.integrated, reduction = &quot;tsne&quot;, group.by = &quot;sex&quot;)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-22-1.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(adult.integrated, reduction = &quot;tsne&quot;, group.by = &quot;batch&quot;)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-23-1.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#FeaturePlot(adult.integrated, features = c(&quot;XIST&quot;))</code></pre>
<pre class="r"><code>FeaturePlot(adult.integrated, features = c(&quot;TNNT2&quot;, &quot;MYH6&quot;, &quot;WT1&quot;, &quot;PECAM1&quot;, &quot;TBX3&quot;, &quot;PDGFRA&quot;), pt.size = 0.2,
            ncol = 3)</code></pre>
<pre><code>Warning: Found the following features in more than one assay, excluding the
default. We will not include these in the final dataframe: PDGFRA</code></pre>
<pre><code>Warning in FetchData(object = object, vars = c(dims, &quot;ident&quot;, features), :
The following requested variables were not found: PDGFRA</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-25-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow=c(1,1))
par(mar=c(4,4,2,2))
tab &lt;- table(Idents(adult.integrated),adult@meta.data$biorep)
barplot(t(tab/rowSums(tab)),beside=TRUE,col=ggplotColors(3),legend=TRUE)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-26-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-26-1.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="visualisation-with-clustree" class="section level1">
<h1>Visualisation with clustree</h1>
<pre class="r"><code>clusres &lt;- c(0.1,0.2,0.3,0.4,0.5,0.6,0.7,0.8,0.9,1,1.1,1.2)</code></pre>
<pre class="r"><code>for(i in 1:length(clusres)){
  adult.integrated &lt;- FindClusters(adult.integrated, 
                                   resolution = clusres[i])
}</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9416
Number of edges: 354612

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9778
Number of communities: 10
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9416
Number of edges: 354612

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9616
Number of communities: 13
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9416
Number of edges: 354612

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9463
Number of communities: 15
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9416
Number of edges: 354612

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9331
Number of communities: 19
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9416
Number of edges: 354612

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9222
Number of communities: 20
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9416
Number of edges: 354612

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9127
Number of communities: 21
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9416
Number of edges: 354612

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9048
Number of communities: 23
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9416
Number of edges: 354612

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8986
Number of communities: 24
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9416
Number of edges: 354612

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8930
Number of communities: 26
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9416
Number of edges: 354612

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8872
Number of communities: 27
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9416
Number of edges: 354612

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8816
Number of communities: 27
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 9416
Number of edges: 354612

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8757
Number of communities: 27
Elapsed time: 0 seconds</code></pre>
<pre class="r"><code>pct.male &lt;- function(x) {mean(x==&quot;m&quot;)}
pct.female &lt;- function(x) {mean(x==&quot;f&quot;)}
biorep1 &lt;- function(x) {mean(x==&quot;a1&quot;)}
biorep2 &lt;- function(x) {mean(x==&quot;a2&quot;)}
biorep3 &lt;- function(x) {mean(x==&quot;a3&quot;)}</code></pre>
<pre class="r"><code>clustree(adult.integrated, prefix = &quot;integrated_snn_res.&quot;)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-29-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>clustree(adult.integrated, prefix = &quot;integrated_snn_res.&quot;,
         node_colour = &quot;biorep&quot;, node_colour_aggr = &quot;biorep1&quot;,assay=&quot;RNA&quot;)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-30-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-30-1.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>clustree(adult.integrated, prefix = &quot;integrated_snn_res.&quot;,
         node_colour = &quot;biorep&quot;, node_colour_aggr = &quot;biorep2&quot;,assay=&quot;RNA&quot;)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-31-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-31-1.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>clustree(adult.integrated, prefix = &quot;integrated_snn_res.&quot;,
         node_colour = &quot;biorep&quot;, node_colour_aggr = &quot;biorep3&quot;,assay=&quot;RNA&quot;)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-32-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-1">
Past versions of unnamed-chunk-32-1.png
</button>
</p>
<div id="fig-unnamed-chunk-32-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-32-1.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>clustree(adult.integrated, prefix = &quot;integrated_snn_res.&quot;,
         node_colour = &quot;sex&quot;, node_colour_aggr = &quot;pct.female&quot;,assay=&quot;RNA&quot;)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-33-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-1">
Past versions of unnamed-chunk-33-1.png
</button>
</p>
<div id="fig-unnamed-chunk-33-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-33-1.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>clustree(adult.integrated, prefix = &quot;integrated_snn_res.&quot;,
         node_colour = &quot;sex&quot;, node_colour_aggr = &quot;pct.male&quot;,assay=&quot;RNA&quot;)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-34-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-34-1">
Past versions of unnamed-chunk-34-1.png
</button>
</p>
<div id="fig-unnamed-chunk-34-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-34-1.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#clustree(adult.integrated, prefix = &quot;integrated_snn_res.&quot;,
#         node_colour = &quot;XIST&quot;, node_colour_aggr = &quot;median&quot;,
#         assay=&quot;RNA&quot;)</code></pre>
<pre class="r"><code>clustree(adult.integrated, prefix = &quot;integrated_snn_res.&quot;,
         node_colour = &quot;TNNT2&quot;, node_colour_aggr = &quot;median&quot;,
         assay=&quot;RNA&quot;)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-36-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="save-seurat-object" class="section level1">
<h1>Save Seurat object</h1>
<pre class="r"><code>DefaultAssay(adult.integrated) &lt;- &quot;RNA&quot;
Idents(adult.integrated) &lt;- adult.integrated$integrated_snn_res.0.6</code></pre>
<pre class="r"><code>saveRDS(adult.integrated,file=&quot;./output/RDataObjects/adult-int.Rds&quot;)</code></pre>
</div>
<div id="find-markers" class="section level1">
<h1>Find Markers</h1>
<pre class="r"><code>par(mfrow=c(1,1))
par(mar=c(4,4,2,2))
tab &lt;- table(Idents(adult.integrated),adult@meta.data$biorep)
barplot(t(tab/rowSums(tab)),beside=TRUE,col=ggplotColors(3),legend=TRUE)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-39-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#adultmarkers &lt;- FindAllMarkers(adult.integrated, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
# Limma-trend for DE

y.adult &lt;- DGEList(alla.keep)

logcounts &lt;- normCounts(y.adult,log=TRUE,prior.count=0.5)

#logcounts.n &lt;- normalizeBetweenArrays(logcounts, method = &quot;cyclicloess&quot;)

maxclust &lt;- length(levels(Idents(adult.integrated)))-1

grp &lt;- paste(&quot;c&quot;,Idents(adult.integrated),sep = &quot;&quot;)
grp &lt;- factor(grp,levels = paste(&quot;c&quot;,0:maxclust,sep=&quot;&quot;))

design &lt;- model.matrix(~0+grp)
colnames(design) &lt;- levels(grp)

mycont &lt;- matrix(NA,ncol=length(levels(grp)),nrow=length(levels(grp)))
rownames(mycont)&lt;-colnames(mycont)&lt;-levels(grp)
diag(mycont)&lt;-1
mycont[upper.tri(mycont)]&lt;- -1/(length(levels(factor(grp)))-1)
mycont[lower.tri(mycont)]&lt;- -1/(length(levels(factor(grp)))-1)

fit &lt;- lmFit(logcounts,design)
fit.cont &lt;- contrasts.fit(fit,contrasts=mycont)
fit.cont &lt;- eBayes(fit.cont,trend=TRUE,robust=TRUE)

fit.cont$genes &lt;- ann.keep

summary(decideTests(fit.cont))</code></pre>
<pre><code>          c0    c1    c2    c3    c4    c5    c6    c7    c8    c9   c10
Down    4054  5600  5412  5159  4219  5439  4471  3932 11588  8258  3068
NotSig  5681  6449  8196  7127  6674  8832  7943  8974  5553  7557 10562
Up      7811  5497  3938  5260  6653  3275  5132  4640   405  1731  3916
         c11   c12   c13   c14   c15   c16   c17   c18   c19   c20
Down    8406  9999  4755  2221  6995  2188  2672   162  1735  1384
NotSig  7542  7129 10254 12838  9474 13024 12804 12408 14126 11703
Up      1598   418  2537  2487  1077  2334  2070  4976  1685  4459</code></pre>
<pre class="r"><code>treat &lt;- treat(fit.cont,lfc=0.5)

dt &lt;- decideTests(treat)
summary(dt)</code></pre>
<pre><code>          c0    c1    c2    c3    c4    c5    c6    c7    c8    c9   c10
Down     314   427   413   572   310   378   364   302  1378   782   285
NotSig 16333 16128 16480 15789 15994 16508 16220 16495 16107 16446 16609
Up       899   991   653  1185  1242   660   962   749    61   318   652
         c11   c12   c13   c14   c15   c16   c17   c18   c19   c20
Down    1412  1232   720   241   801   253   306     0   156   222
NotSig 15716 16217 16247 16735 16427 16663 16696 16877 16928 16191
Up       418    97   579   570   318   630   544   669   462  1133</code></pre>
<pre class="r"><code>par(mfrow=c(3,3))
for(i in 1:ncol(mycont)){
  plotMD(treat,coef=i,status = dt[,i],hl.cex=0.5)
  abline(h=0,col=colours()[c(226)])
  lines(lowess(treat$Amean,treat$coefficients[,i]),lwd=1.5,col=4)
}</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-40-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-40-1">
Past versions of unnamed-chunk-40-1.png
</button>
</p>
<div id="fig-unnamed-chunk-40-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/b502e303c4649847e0126a8cf756e0c2da85bf08/public/figure/04-ClusterAdult.Rmd/unnamed-chunk-40-1.png" target="_blank">b502e30</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-07-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-40-2.png" width="960" style="display: block; margin: auto;" /><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-40-3.png" width="960" style="display: block; margin: auto;" /></p>
<div id="write-out-marker-genes-for-each-cluster" class="section level2">
<h2>Write out marker genes for each cluster</h2>
<pre class="r"><code>#write.csv(adultmarkers,file=&quot;./output/AllAdult-clustermarkers.csv&quot;)

contnames &lt;- colnames(mycont)

for(i in 1:length(contnames)){
  topsig &lt;- topTreat(treat,coef=i,n=Inf,p.value=0.05)
  write.csv(topsig[topsig$logFC&gt;0,],file=paste(&quot;./output/MarkerAnalysis/Adult/Up-Cluster-&quot;,contnames[i],&quot;.csv&quot;,sep=&quot;&quot;))
  write.csv(topGO(goana(de=topsig$ENTREZID[topsig$logFC&gt;0],universe=treat$genes$ENTREZID,species=&quot;Hs&quot;),number=50),
            file=paste(&quot;./output/MarkerAnalysis/Adult/GO-Cluster-&quot;,contnames[i],&quot;.csv&quot;,sep=&quot;&quot;))

}</code></pre>
</div>
</div>
<div id="heatmap-of-pre-defined-heart-cell-type-markers" class="section level1">
<h1>Heatmap of pre-defined heart cell type markers</h1>
<pre class="r"><code>hm &lt;- read.delim(&quot;./data/heart-markers-long.txt&quot;,stringsAsFactors = FALSE)
hgene &lt;- toupper(hm$Gene)
hgene &lt;- unique(hgene)

m &lt;- match(hgene,rownames(logcounts))
m &lt;- m[!is.na(m)]

sam &lt;- factor(adult.integrated$biorep)
newgrp &lt;- paste(grp,sam,sep=&quot;.&quot;)
newgrp &lt;- factor(newgrp,levels=paste(rep(levels(grp),each=3),levels(sam),sep=&quot;.&quot;))
o &lt;-order(newgrp)

annot &lt;- data.frame(CellType=grp,Sample=sam,NewGroup=newgrp)

mycelltypes &lt;- hm$Celltype[match(rownames(logcounts)[m],toupper(hm$Gene))]
mycelltypes &lt;- factor(mycelltypes)


myColors &lt;- list(Clust=NA,Sample=NA,Celltypes=NA)
myColors$Clust&lt;-sample(ggplotColors(length(levels(grp))),length(levels(grp)))
names(myColors$Clust)&lt;-levels(grp)
myColors$Sample &lt;- brewer.pal(3, &quot;Set1&quot;)
names(myColors$Sample) &lt;- levels(sam)
myColors$Celltypes &lt;- ggplotColors(length(levels(mycelltypes)))
names(myColors$Celltypes) &lt;- levels(mycelltypes) 

mygenes &lt;- rownames(logcounts)[m]
mygenelab &lt;- paste(mygenes,mycelltypes,sep=&quot;_&quot;)</code></pre>
<pre class="r"><code>par(mfrow=c(1,1))
pdf(file=&quot;./output/Figures/adult-heatmap-hmarkers.pdf&quot;,width=20,height=15,onefile = FALSE)
aheatmap(logcounts[m,o],Rowv=NA,Colv=NA,labRow=mygenelab,labCol=NA,
         annCol=list(Clust=grp[o],Sample=sam[o]),
#         annRow = list(Celltypes=mycelltypes),
         annColors = myColors, 
         fontsize=16,color=&quot;-RdYlBu&quot;)
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
</div>
<div id="summarise-expression-across-cells" class="section level1">
<h1>Summarise expression across cells</h1>
<pre class="r"><code>sumexpr &lt;- matrix(NA,nrow=nrow(logcounts),ncol=length(levels(newgrp)))
rownames(sumexpr) &lt;- rownames(logcounts)
colnames(sumexpr) &lt;- levels(newgrp)

for(i in 1:nrow(sumexpr)){
  sumexpr[i,] &lt;- tapply(logcounts[i,],newgrp,median)
}</code></pre>
<pre class="r"><code>par(mfrow=c(1,1))
clust &lt;- rep(levels(grp),each=3)
samps &lt;- rep(levels(sam),length(levels(grp)))
pdf(file=&quot;./output/Figures/adult-heatmap-hmarkers-summarised.pdf&quot;,width=20,height=15,onefile = FALSE)
aheatmap(sumexpr[m,],Rowv = NA,Colv = NA, labRow = mygenelab,
         annCol=list(Clust=clust,Sample=samps),
 #        annRow=list(Celltypes=mycelltypes),
         annColors=myColors,
         fontsize=14,color=&quot;-RdYlBu&quot;)
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>aheatmap(sumexpr[m,],Rowv = NA,Colv = NA, labRow = mygenelab,
         annCol=list(Clust=clust,Sample=samps),
#         annRow=list(Celltypes=mycelltypes),
         annColors=myColors,
         fontsize=12,color=&quot;-RdYlBu&quot;)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/unnamed-chunk-45-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sig.genes &lt;- gene.label &lt;- vector(&quot;list&quot;, length(levels(grp)))
for(i in 1:length(sig.genes)){
  top &lt;- topTreat(treat,coef=i,n=200)
  sig.genes[[i]] &lt;- rownames(top)[top$logFC&gt;0][1:5]
  gene.label[[i]] &lt;- paste(rownames(top)[top$logFC&gt;0][1:5],levels(grp)[i],sep=&quot;-&quot;)
} 

csig &lt;- unlist(sig.genes)
genes &lt;- unlist(gene.label)

pdf(file=&quot;./output/Figures/adult-heatmap-siggenes-summarised.pdf&quot;,width=20,height=15,onefile = FALSE)
aheatmap(sumexpr[csig,],Rowv = NA,Colv = NA, labRow = genes,
         annCol=list(Clust=clust,Sample=samps),
         annColors=myColors,
         fontsize=16,color=&quot;-RdYlBu&quot;,
         scale=&quot;none&quot;)
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>aheatmap(sumexpr[csig,],Rowv = NA,Colv = NA, labRow = genes,
         annCol=list(Clust=clust,Sample=samps),
         annColors=myColors,
         fontsize=16,color=&quot;-RdYlBu&quot;,
         scale=&quot;none&quot;)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/heatmap-sig-genes-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="try-annotate-clusters-based-on-fetal-cell-types" class="section level1">
<h1>Try annotate clusters based on fetal cell types</h1>
<pre class="r"><code># This loads a list object called sig.genes.gst
load(file=&quot;./data/gstlist-fetal.Rdata&quot;)

expr.score &lt;- matrix(NA,nrow=length(sig.genes.gst),ncol=length(levels(newgrp)))
colnames(expr.score) &lt;- levels(newgrp)
rownames(expr.score) &lt;- names(sig.genes.gst)

specialm &lt;- lapply(sig.genes.gst,function(x) match(x,rownames(sumexpr))[!is.na(match(x,rownames(sumexpr)))])

for(i in 1:nrow(expr.score)){
  expr.score[i,] &lt;- colMedians(sumexpr[specialm[[i]],])  
}

pdf(file=&quot;./output/Figures/heatmap-match-fetal-adult-medians.pdf&quot;,width=20,height=15,onefile = FALSE)
aheatmap(expr.score,
         Rowv = NA,Colv = NA, 
         labRow = rownames(expr.score),
         annCol=list(Clust=clust,Sample=samps),
         annColors=myColors,
         fontsize=16,color=&quot;-RdYlBu&quot;,
         scale=&quot;none&quot;)
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>aheatmap(expr.score,
         Rowv = NA,Colv = NA, 
         labRow = rownames(expr.score),
         annCol=list(Clust=clust,Sample=samps),
         annColors=myColors,
         fontsize=16,color=&quot;-RdYlBu&quot;,
         scale=&quot;none&quot;)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/heatmap-match-fetal-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="compare-adult-to-young" class="section level1">
<h1>Compare adult to young</h1>
<pre class="r"><code># This loads a list object called young.sig.genes.gst
load(file=&quot;./data/gstlist-young.Rdata&quot;)

expr.score &lt;- matrix(NA,nrow=length(young.sig.genes.gst),ncol=length(levels(newgrp)))
colnames(expr.score) &lt;- levels(newgrp)
rownames(expr.score) &lt;- names(young.sig.genes.gst)

specialm &lt;- lapply(young.sig.genes.gst,function(x) match(x,rownames(sumexpr))[!is.na(match(x,rownames(sumexpr)))])

for(i in 1:nrow(expr.score)){
  expr.score[i,] &lt;- colMedians(sumexpr[specialm[[i]],])  
}

pdf(file=&quot;./output/Figures/heatmap-match-young-adult-medians.pdf&quot;,width=20,height=15,onefile = FALSE)
aheatmap(expr.score,
         Rowv = NA,Colv = NA, 
         labRow = rownames(expr.score),
         annCol=list(Clust=clust,Sample=samps),
         annColors=myColors,
         fontsize=16,color=&quot;-RdYlBu&quot;,
         scale=&quot;none&quot;)
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>aheatmap(expr.score,
         Rowv = NA,Colv = NA, 
         labRow = rownames(expr.score),
         annCol=list(Clust=clust,Sample=samps),
         annColors=myColors,
         fontsize=16,color=&quot;-RdYlBu&quot;,
         scale=&quot;none&quot;)</code></pre>
<p><img src="figure/04-ClusterAdult.Rmd/heatmap-match-young-1.png" width="960" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.0 (2019-04-26)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS release 6.7 (Final)

Matrix products: default
BLAS:   /usr/local/installed/R/3.6.0/lib64/R/lib/libRblas.so
LAPACK: /usr/local/installed/R/3.6.0/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] splines   parallel  stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] dplyr_0.8.1                 clustree_0.4.0             
 [3] ggraph_1.0.2                workflowr_1.3.0            
 [5] NMF_0.21.0                  bigmemory_4.5.33           
 [7] cluster_2.0.9               rngtools_1.3.1.1           
 [9] pkgmaker_0.27               registry_0.5-1             
[11] scran_1.12.0                SingleCellExperiment_1.6.0 
[13] SummarizedExperiment_1.14.0 GenomicRanges_1.36.0       
[15] GenomeInfoDb_1.20.0         DelayedArray_0.10.0        
[17] BiocParallel_1.18.0         matrixStats_0.54.0         
[19] cowplot_0.9.4               monocle_2.12.0             
[21] DDRTree_0.1.5               irlba_2.3.3                
[23] VGAM_1.1-1                  ggplot2_3.1.1              
[25] Matrix_1.2-17               Seurat_3.0.2               
[27] org.Hs.eg.db_3.8.2          AnnotationDbi_1.46.0       
[29] IRanges_2.18.0              S4Vectors_0.22.0           
[31] Biobase_2.44.0              BiocGenerics_0.30.0        
[33] RColorBrewer_1.1-2          edgeR_3.26.3               
[35] limma_3.40.2               

loaded via a namespace (and not attached):
  [1] reticulate_1.12          R.utils_2.8.0           
  [3] tidyselect_0.2.5         RSQLite_2.1.1           
  [5] htmlwidgets_1.3          grid_3.6.0              
  [7] combinat_0.0-8           docopt_0.6.1            
  [9] Rtsne_0.15               munsell_0.5.0           
 [11] codetools_0.2-16         ica_1.0-2               
 [13] statmod_1.4.30           future_1.12.0           
 [15] withr_2.1.2              colorspace_1.4-1        
 [17] fastICA_1.2-1            knitr_1.22              
 [19] ROCR_1.0-7               gbRd_0.4-11             
 [21] listenv_0.7.0            Rdpack_0.11-0           
 [23] labeling_0.3             git2r_0.25.2            
 [25] slam_0.1-45              GenomeInfoDbData_1.2.1  
 [27] polyclip_1.10-0          bit64_0.9-7             
 [29] farver_1.1.0             pheatmap_1.0.12         
 [31] rprojroot_1.3-2          xfun_0.6                
 [33] R6_2.4.0                 doParallel_1.0.14       
 [35] ggbeeswarm_0.6.0         rsvd_1.0.0              
 [37] locfit_1.5-9.1           bitops_1.0-6            
 [39] assertthat_0.2.1         SDMTools_1.1-221.1      
 [41] scales_1.0.0             beeswarm_0.2.3          
 [43] gtable_0.3.0             npsurv_0.4-0            
 [45] globals_0.12.4           tidygraph_1.1.2         
 [47] rlang_0.3.4              lazyeval_0.2.2          
 [49] checkmate_1.9.3          yaml_2.2.0              
 [51] reshape2_1.4.3           backports_1.1.4         
 [53] tools_3.6.0              gridBase_0.4-7          
 [55] gplots_3.0.1.1           dynamicTreeCut_1.63-1   
 [57] ggridges_0.5.1           Rcpp_1.0.1              
 [59] plyr_1.8.4               zlibbioc_1.30.0         
 [61] purrr_0.3.2              RCurl_1.95-4.12         
 [63] densityClust_0.3         pbapply_1.4-0           
 [65] viridis_0.5.1            zoo_1.8-5               
 [67] ggrepel_0.8.1            fs_1.3.1                
 [69] magrittr_1.5             data.table_1.11.6       
 [71] lmtest_0.9-37            RANN_2.6.1              
 [73] whisker_0.3-2            fitdistrplus_1.0-14     
 [75] lsei_1.2-0               evaluate_0.13           
 [77] xtable_1.8-4             sparsesvd_0.1-4         
 [79] gridExtra_2.3            HSMMSingleCell_1.4.0    
 [81] compiler_3.6.0           scater_1.12.2           
 [83] tibble_2.1.1             KernSmooth_2.23-15      
 [85] crayon_1.3.4             R.oo_1.22.0             
 [87] htmltools_0.3.6          tidyr_0.8.3             
 [89] DBI_1.0.0                tweenr_1.0.1            
 [91] MASS_7.3-51.4            R.methodsS3_1.7.1       
 [93] gdata_2.18.0             metap_1.1               
 [95] igraph_1.2.4.1           pkgconfig_2.0.2         
 [97] bigmemory.sri_0.1.3      plotly_4.9.0            
 [99] foreach_1.4.4            vipor_0.4.5             
[101] dqrng_0.2.1              XVector_0.24.0          
[103] bibtex_0.4.2             stringr_1.4.0           
[105] digest_0.6.18            sctransform_0.2.0       
[107] tsne_0.1-3               rmarkdown_1.12.6        
[109] DelayedMatrixStats_1.6.0 gtools_3.8.1            
[111] nlme_3.1-139             jsonlite_1.6            
[113] BiocNeighbors_1.2.0      viridisLite_0.3.0       
[115] pillar_1.3.1             lattice_0.20-38         
[117] GO.db_3.8.2              httr_1.4.0              
[119] survival_2.44-1.1        glue_1.3.1              
[121] qlcMatrix_0.9.7          FNN_1.1.3               
[123] png_0.1-7                iterators_1.0.10        
[125] bit_1.1-14               ggforce_0.2.2           
[127] stringi_1.4.3            blob_1.1.1              
[129] BiocSingular_1.0.0       caTools_1.17.1.2        
[131] memoise_1.1.0            future.apply_1.2.0      
[133] ape_5.3                 </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
