<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Belinda Phipson" />


<title>Recluster epicardial lineage</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Porello-heart-snRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://git.mcri.edu.au/bioinformatics/Porello-heart-snRNAseq">
    <span class="fa fa-gitlab"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Recluster epicardial lineage</h1>
<h4 class="author">Belinda Phipson</h4>
<h4 class="date">8/15/2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-10-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Porello-heart-snRNAseq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.3.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190603code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190603)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190603code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190603)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgitmcrieduaubioinformaticsporelloheartsnrnaseqtreea1cc8330f8dc9b905d8c31c2a8c0b52c0b99e504targetblanka1cc833a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/tree/a1cc8330f8dc9b905d8c31c2a8c0b52c0b99e504" target="_blank">a1cc833</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgitmcrieduaubioinformaticsporelloheartsnrnaseqtreea1cc8330f8dc9b905d8c31c2a8c0b52c0b99e504targetblanka1cc833a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/07b-DECardioFYA.Rmd
    Untracked:  analysis/PBapproachMarker-cardio.Rmd
    Untracked:  analysis/doublet-detection.Rmd
    Untracked:  broad_props.csv
    Untracked:  code/ReadDataObjects.R
    Untracked:  code/getTransformedProps.R
    Untracked:  data/adult-clust.txt
    Untracked:  data/dcm-clust.txt
    Untracked:  data/fetal-clust.txt
    Untracked:  data/gstlist-adult.Rdata
    Untracked:  data/gstlist-dcm-res03.Rdata
    Untracked:  data/gstlist-dcm.Rdata
    Untracked:  data/gstlist-fetal.Rdata
    Untracked:  data/gstlist-young.Rdata
    Untracked:  data/heart-markers-long.txt
    Untracked:  data/immune-markers-long.txt
    Untracked:  data/pseudobulk-pool.Rds
    Untracked:  data/pseudobulk.Rds
    Untracked:  data/targets_pools.txt
    Untracked:  data/young-clust.txt
    Untracked:  output/AllAdult-clustermarkers-v2.csv
    Untracked:  output/AllAdult-clustermarkers.csv
    Untracked:  output/AllFetal-clustermarkers.csv
    Untracked:  output/AllYoung-clustermarkers.csv
    Untracked:  output/Alldcm-clustermarkers.csv
    Untracked:  output/DEAnalysis/
    Untracked:  output/Figures/
    Untracked:  output/MarkerAnalysis/
    Untracked:  output/RDataObjects/
    Untracked:  output/cardio-numcells-clusters.csv
    Untracked:  output/cardio-numcells-clusters.txt
    Untracked:  output/fetal1-clustermarkers.csv
    Untracked:  output/fetal2-clustermarkers.csv
    Untracked:  output/fetal3-clustermarkers.csv
    Untracked:  output/heatmap-top10-adultmarkergenes.pdf
    Untracked:  output/young1-clustermarkers.csv

Unstaged changes:
    Modified:   analysis/01-QualityControl.Rmd
    Modified:   analysis/01a-DEpseudobulk.Rmd
    Modified:   analysis/02-ClusterFetal.Rmd
    Modified:   analysis/02c-ClusterFetal3.Rmd
    Modified:   analysis/03-ClusterYoung.Rmd
    Modified:   analysis/04-ClusterAdult.Rmd
    Modified:   analysis/07a-ClustCardioFYA.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://git.mcri.edu.au/bioinformatics/porello-heart-snrnaseq/blob/a1cc8330f8dc9b905d8c31c2a8c0b52c0b99e504/analysis/08-ClustEpicardial.Rmd" target="_blank">a1cc833</a>
</td>
<td>
Belinda Phipson
</td>
<td>
2019-10-28
</td>
<td>
recluster epicardial lineage
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries-and-functions" class="section level1">
<h1>Load libraries and functions</h1>
<pre class="r"><code>library(edgeR)
library(RColorBrewer)
library(org.Hs.eg.db)
library(limma)
library(Seurat)
library(monocle)
library(cowplot)
library(DelayedArray)
library(scran)
library(NMF)
library(workflowr)
library(ggplot2)
library(clustree)
library(dplyr)</code></pre>
<pre class="r"><code>source(&quot;/misc/card2-single_cell_nuclei_rnaseq/Porello-heart-snRNAseq/code/normCounts.R&quot;)
source(&quot;/misc/card2-single_cell_nuclei_rnaseq/Porello-heart-snRNAseq/code/findModes.R&quot;)
source(&quot;/misc/card2-single_cell_nuclei_rnaseq/Porello-heart-snRNAseq/code/ggplotColors.R&quot;)</code></pre>
<pre class="r"><code>targets &lt;- read.delim(&quot;/misc/card2-single_cell_nuclei_rnaseq/Porello-heart-snRNAseq/data/targets.txt&quot;,header=TRUE, stringsAsFactors = FALSE)
targets$FileName2 &lt;- paste(targets$FileName,&quot;/&quot;,sep=&quot;&quot;)
targets$Group_ID2 &lt;- gsub(&quot;LV_&quot;,&quot;&quot;,targets$Group_ID)
group &lt;- c(&quot;Fetal_1&quot;,&quot;Fetal_2&quot;,&quot;Fetal_3&quot;,
           &quot;Young_1&quot;,&quot;Young_2&quot;,&quot;Young_3&quot;,
           &quot;Adult_1&quot;,&quot;Adult_2&quot;,&quot;Adult_3&quot;, 
           &quot;Diseased_1&quot;,&quot;Diseased_2&quot;,
           &quot;Diseased_3&quot;,&quot;Diseased_4&quot;)
m &lt;- match(group, targets$Group_ID2)
targets &lt;- targets[m,]</code></pre>
<pre class="r"><code>fetal.integrated &lt;- readRDS(file=&quot;./output/RDataObjects/fetal-int.Rds&quot;)
load(file=&quot;./output/RDataObjects/fetalObjs.Rdata&quot;)

young.integrated &lt;- readRDS(file=&quot;./output/RDataObjects/young-int.Rds&quot;)
load(file=&quot;./output/RDataObjects/youngObjs.Rdata&quot;)

adult.integrated &lt;- readRDS(file=&quot;./output/RDataObjects/adult-int.Rds&quot;)
load(file=&quot;./output/RDataObjects/adultObjs.Rdata&quot;)</code></pre>
</div>
<div id="set-default-clustering-resolution" class="section level1">
<h1>Set default clustering resolution</h1>
<pre class="r"><code># Default 0.3
Idents(fetal.integrated) &lt;- fetal.integrated$integrated_snn_res.0.3
DimPlot(fetal.integrated, reduction = &quot;tsne&quot;,label=TRUE,label.size = 6)+NoLegend()</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Default 0.3
DimPlot(young.integrated, reduction = &quot;tsne&quot;,label=TRUE,label.size = 6)+NoLegend()</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Default 0.6
DimPlot(adult.integrated, reduction = &quot;tsne&quot;,label=TRUE,label.size = 6)+NoLegend()</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="merge-all-data-together" class="section level1">
<h1>Merge all data together</h1>
<pre class="r"><code>heart &lt;- merge(fetal.integrated, y = c(young.integrated, adult.integrated), project = &quot;heart&quot;)
table(heart$orig.ident)</code></pre>
<pre><code>
adult fetal young 
 9416 27760 16964 </code></pre>
<pre class="r"><code>DefaultAssay(object = heart) &lt;- &quot;RNA&quot;</code></pre>
</div>
<div id="get-epicardial-cells-only" class="section level1">
<h1>Get epicardial cells only</h1>
<pre class="r"><code>epic &lt;- subset(heart,subset = Broad_celltype == &quot;Epicardial cells&quot;)</code></pre>
</div>
<div id="filter-out-crappy-cells" class="section level1">
<h1>Filter out crappy cells</h1>
<p>Check for cells with very low number of uniquely detected genes.</p>
<pre class="r"><code>par(mfrow=c(1,2))
plot(density(epic$nFeature_RNA),main=&quot;Number of genes detected&quot;)
abline(v=500,col=2)
plot(density(epic$nCount_RNA),main=&quot;Library size&quot;)
abline(v=2500,col=2)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#epic &lt;- subset(epic, subset = nFeature_RNA &gt; 500 &amp; nCount_RNA &gt; 2500)
dim(epic)</code></pre>
<pre><code>[1] 17926  3474</code></pre>
<pre class="r"><code>table(epic$biorep)</code></pre>
<pre><code>
 a1  a2  a3  f1  f2  f3  y1  y2  y3 
343 493  92 564 425 404 613 280 260 </code></pre>
</div>
<div id="run-new-integration-with-sctransform-normalisation" class="section level1">
<h1>Run new integration with SCtransform normalisation</h1>
<pre class="r"><code>epic.list &lt;- SplitObject(epic, split.by = &quot;biorep&quot;)</code></pre>
<pre class="r"><code>for (i in 1:length(epic.list)) {
    epic.list[[i]] &lt;- SCTransform(epic.list[[i]], verbose = FALSE)
}</code></pre>
<pre class="r"><code>min(sapply(epic.list, ncol))</code></pre>
<pre><code>[1] 92</code></pre>
<pre class="r"><code>epic.anchors &lt;- FindIntegrationAnchors(object.list = epic.list, dims=1:30,anchor.features = 3000,k.filter=92)
epic.integrated &lt;- IntegrateData(anchorset = epic.anchors,dims=1:30)</code></pre>
</div>
<div id="perform-clustering" class="section level1">
<h1>Perform clustering</h1>
<pre class="r"><code>DefaultAssay(object = epic.integrated) &lt;- &quot;integrated&quot;</code></pre>
<div id="perform-scaling-and-pca" class="section level2">
<h2>Perform scaling and PCA</h2>
<pre class="r"><code>epic.integrated &lt;- ScaleData(epic.integrated, verbose = FALSE)
epic.integrated &lt;- RunPCA(epic.integrated, npcs = 50, verbose = FALSE)
ElbowPlot(epic.integrated,ndims=50)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/pca-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VizDimLoadings(epic.integrated, dims = 1:4, reduction = &quot;pca&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-10-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(epic.integrated, reduction = &quot;pca&quot;,group.by=&quot;orig.ident&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(epic.integrated, reduction = &quot;pca&quot;,group.by=&quot;biorep&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(epic.integrated, reduction = &quot;pca&quot;,group.by=&quot;sex&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-11-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(epic.integrated, reduction = &quot;pca&quot;,group.by=&quot;batch&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-11-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimHeatmap(epic.integrated, dims = 1:15, cells = 500, balanced = TRUE)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-12-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimHeatmap(epic.integrated, dims = 16:30, cells = 500, balanced = TRUE)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-12-2.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimHeatmap(epic.integrated, dims = 31:45, cells = 500, balanced = TRUE)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-12-3.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="perform-nearest-neighbours-clustering" class="section level2">
<h2>Perform nearest neighbours clustering</h2>
<pre class="r"><code>epic.integrated &lt;- FindNeighbors(epic.integrated, dims = 1:20)
epic.integrated &lt;- FindClusters(epic.integrated, resolution = 0.1)</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3474
Number of edges: 158223

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9202
Number of communities: 4
Elapsed time: 0 seconds</code></pre>
<pre class="r"><code>table(Idents(epic.integrated))</code></pre>
<pre><code>
   0    1    2    3 
2370  832  192   80 </code></pre>
<pre class="r"><code>par(mar=c(5,4,2,2))
barplot(table(Idents(epic.integrated)),ylab=&quot;Number of cells&quot;,xlab=&quot;Clusters&quot;)
title(&quot;Number of cells in each cluster&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="visualisation-with-tsne" class="section level2">
<h2>Visualisation with TSNE</h2>
<pre class="r"><code>set.seed(10)
epic.integrated &lt;- RunTSNE(epic.integrated, reduction = &quot;pca&quot;, dims = 1:20)</code></pre>
<pre class="r"><code>DimPlot(epic.integrated, reduction = &quot;tsne&quot;,label=TRUE,label.size = 6,pt.size = 0.5)+NoLegend()</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pdf(file=&quot;./output/Figures/tsne-epicALL-res01.pdf&quot;,width=10,height=8,onefile = FALSE)
DimPlot(epic.integrated, reduction = &quot;tsne&quot;,label=TRUE,label.size = 6,pt.size = 0.5)+NoLegend()
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>DimPlot(epic.integrated, reduction = &quot;tsne&quot;, group.by = &quot;orig.ident&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(epic.integrated, reduction = &quot;tsne&quot;, split.by = &quot;orig.ident&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(epic.integrated, reduction = &quot;tsne&quot;, group.by = &quot;biorep&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(epic.integrated, reduction = &quot;tsne&quot;, group.by = &quot;sex&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(epic.integrated, reduction = &quot;tsne&quot;, split.by = &quot;sex&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(epic.integrated, reduction = &quot;tsne&quot;, group.by = &quot;batch&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow=c(1,1))
par(mar=c(4,4,2,2))
tab &lt;- table(Idents(epic.integrated),epic.integrated$biorep)
barplot(t(tab/rowSums(tab)),beside=TRUE,col=ggplotColors(9),legend=TRUE)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-22-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow=c(1,1))
par(mar=c(4,4,2,2))
tab &lt;- table(Idents(epic.integrated),epic.integrated$orig.ident)
barplot(t(tab/rowSums(tab)),beside=TRUE,col=ggplotColors(3))
legend(&quot;topleft&quot;,legend=colnames(tab),fill=ggplotColors(3))</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-23-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="visualisation-with-clustree" class="section level1">
<h1>Visualisation with clustree</h1>
<pre class="r"><code>clusres &lt;- c(0.1,0.2,0.3,0.4,0.5,0.6,0.7,0.8,0.9,1,1.1,1.2)</code></pre>
<pre class="r"><code>for(i in 1:length(clusres)){
  epic.integrated &lt;- FindClusters(epic.integrated, 
                                   resolution = clusres[i])
}</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3474
Number of edges: 158223

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9202
Number of communities: 4
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3474
Number of edges: 158223

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8777
Number of communities: 6
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3474
Number of edges: 158223

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8464
Number of communities: 7
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3474
Number of edges: 158223

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8172
Number of communities: 8
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3474
Number of edges: 158223

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7924
Number of communities: 9
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3474
Number of edges: 158223

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7687
Number of communities: 9
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3474
Number of edges: 158223

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7466
Number of communities: 9
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3474
Number of edges: 158223

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7319
Number of communities: 11
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3474
Number of edges: 158223

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7188
Number of communities: 11
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3474
Number of edges: 158223

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7059
Number of communities: 12
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3474
Number of edges: 158223

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.6935
Number of communities: 13
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3474
Number of edges: 158223

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.6836
Number of communities: 13
Elapsed time: 0 seconds</code></pre>
<pre class="r"><code>pct.male &lt;- function(x) {mean(x==&quot;m&quot;)}
pct.female &lt;- function(x) {mean(x==&quot;f&quot;)}
pct.fetal &lt;- function(x) {mean(x==&quot;fetal&quot;)}
pct.young &lt;- function(x) {mean(x==&quot;young&quot;)}
pct.adult &lt;- function(x) {mean(x==&quot;adult&quot;)}</code></pre>
<pre class="r"><code>clustree(epic.integrated, prefix = &quot;integrated_snn_res.&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-26-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>clustree(epic.integrated, prefix = &quot;integrated_snn_res.&quot;,
         node_colour = &quot;sex&quot;, node_colour_aggr = &quot;pct.female&quot;,assay=&quot;RNA&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-27-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>clustree(epic.integrated, prefix = &quot;integrated_snn_res.&quot;,
         node_colour = &quot;sex&quot;, node_colour_aggr = &quot;pct.male&quot;,assay=&quot;RNA&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-28-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>clustree(epic.integrated, prefix = &quot;integrated_snn_res.&quot;,
         node_colour = &quot;orig.ident&quot;, node_colour_aggr = &quot;pct.fetal&quot;,assay=&quot;RNA&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-29-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>clustree(epic.integrated, prefix = &quot;integrated_snn_res.&quot;,
         node_colour = &quot;orig.ident&quot;, node_colour_aggr = &quot;pct.young&quot;,assay=&quot;RNA&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-30-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>clustree(epic.integrated, prefix = &quot;integrated_snn_res.&quot;,
         node_colour = &quot;orig.ident&quot;, node_colour_aggr = &quot;pct.adult&quot;,assay=&quot;RNA&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-31-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="save-seurat-object" class="section level1">
<h1>Save Seurat object</h1>
<pre class="r"><code>DefaultAssay(epic.integrated) &lt;- &quot;RNA&quot;
Idents(epic.integrated) &lt;- epic.integrated$integrated_snn_res.0.1</code></pre>
<pre class="r"><code>saveRDS(epic.integrated,file=&quot;./output/RDataObjects/epic-int-FYA-filtered.Rds&quot;)
#epic.integrated &lt;- readRDS(file=&quot;./output/RDataObjects/epic-int-FYA.Rds&quot;)
# Load unfiltered counts matrix for every sample (object all)
load(&quot;./output/RDataObjects/all-counts.Rdata&quot;)</code></pre>
<pre class="r"><code>columns(org.Hs.eg.db)</code></pre>
<pre><code> [1] &quot;ACCNUM&quot;       &quot;ALIAS&quot;        &quot;ENSEMBL&quot;      &quot;ENSEMBLPROT&quot; 
 [5] &quot;ENSEMBLTRANS&quot; &quot;ENTREZID&quot;     &quot;ENZYME&quot;       &quot;EVIDENCE&quot;    
 [9] &quot;EVIDENCEALL&quot;  &quot;GENENAME&quot;     &quot;GO&quot;           &quot;GOALL&quot;       
[13] &quot;IPI&quot;          &quot;MAP&quot;          &quot;OMIM&quot;         &quot;ONTOLOGY&quot;    
[17] &quot;ONTOLOGYALL&quot;  &quot;PATH&quot;         &quot;PFAM&quot;         &quot;PMID&quot;        
[21] &quot;PROSITE&quot;      &quot;REFSEQ&quot;       &quot;SYMBOL&quot;       &quot;UCSCKG&quot;      
[25] &quot;UNIGENE&quot;      &quot;UNIPROT&quot;     </code></pre>
<pre class="r"><code>ann &lt;- AnnotationDbi:::select(org.Hs.eg.db,keys=rownames(all),columns=c(&quot;SYMBOL&quot;,&quot;ENTREZID&quot;,&quot;ENSEMBL&quot;,&quot;GENENAME&quot;,&quot;CHR&quot;),keytype = &quot;SYMBOL&quot;)
m &lt;- match(rownames(all),ann$SYMBOL)
ann &lt;- ann[m,]
table(ann$SYMBOL==rownames(all))</code></pre>
<pre><code>
 TRUE 
33939 </code></pre>
<pre class="r"><code>mito &lt;- grep(&quot;mitochondrial&quot;,ann$GENENAME)
length(mito)</code></pre>
<pre><code>[1] 226</code></pre>
<pre class="r"><code>ribo &lt;- grep(&quot;ribosomal&quot;,ann$GENENAME)
length(ribo)</code></pre>
<pre><code>[1] 198</code></pre>
<pre class="r"><code>missingEZID &lt;- which(is.na(ann$ENTREZID))
length(missingEZID)</code></pre>
<pre><code>[1] 10530</code></pre>
</div>
<div id="find-markers" class="section level1">
<h1>Find Markers</h1>
<pre class="r"><code># Limma-trend for DE
m &lt;- match(colnames(epic.integrated),colnames(all))
all.counts &lt;- all[,m]</code></pre>
<pre class="r"><code>chuck &lt;- unique(c(mito,ribo,missingEZID))
length(chuck)</code></pre>
<pre><code>[1] 10875</code></pre>
<pre class="r"><code>all.counts.keep &lt;- all.counts[-chuck,]
ann.keep &lt;- ann[-chuck,]
table(ann.keep$SYMBOL==rownames(all.counts.keep))</code></pre>
<pre><code>
 TRUE 
23064 </code></pre>
<pre class="r"><code>numzero.genes &lt;- rowSums(all.counts.keep==0)

#avg.exp &lt;- rowMeans(cpm.DGEList(y.kid,log=TRUE))

#plot(avg.exp,numzero.genes,xlab=&quot;Average log-normalised-counts&quot;,ylab=&quot;Number zeroes per gene&quot;)

table(numzero.genes &gt; (ncol(all.counts.keep)-20))</code></pre>
<pre><code>
FALSE  TRUE 
15402  7662 </code></pre>
<pre class="r"><code>keep.genes &lt;- numzero.genes &lt; (ncol(all.counts.keep)-20)
table(keep.genes)</code></pre>
<pre><code>keep.genes
FALSE  TRUE 
 7760 15304 </code></pre>
<pre class="r"><code>all.keep &lt;- all.counts.keep[keep.genes,]
dim(all.keep)</code></pre>
<pre><code>[1] 15304  3474</code></pre>
<pre class="r"><code>ann.keep &lt;- ann.keep[keep.genes,]</code></pre>
<pre class="r"><code>y.epic &lt;- DGEList(all.keep)

logcounts &lt;- normCounts(y.epic,log=TRUE,prior.count=0.5)

#logcounts.n &lt;- normalizeBetweenArrays(logcounts, method = &quot;cyclicloess&quot;)

maxclust &lt;- length(levels(Idents(epic.integrated)))-1

grp &lt;- paste(&quot;c&quot;,Idents(epic.integrated),sep = &quot;&quot;)
grp &lt;- factor(grp,levels = paste(&quot;c&quot;,0:maxclust,sep=&quot;&quot;))

design &lt;- model.matrix(~0+grp+epic.integrated$biorep)
colnames(design)[1:(maxclust+1)] &lt;- levels(grp)

mycont &lt;- matrix(0,ncol=length(levels(grp)),nrow=length(levels(grp)))
colnames(mycont)&lt;-levels(grp)
diag(mycont)&lt;-1
mycont[upper.tri(mycont)]&lt;- -1/(length(levels(factor(grp)))-1)
mycont[lower.tri(mycont)]&lt;- -1/(length(levels(factor(grp)))-1)

# Fill out remaining rows with 0s
zero.rows &lt;- matrix(0,ncol=length(levels(grp)),nrow=(ncol(design)-length(levels(Idents(epic.integrated)))))
test &lt;- rbind(mycont,zero.rows)


fit &lt;- lmFit(logcounts,design)
fit.cont &lt;- contrasts.fit(fit,contrasts=test)
fit.cont &lt;- eBayes(fit.cont,trend=TRUE,robust=TRUE)

fit.cont$genes &lt;- ann.keep

summary(decideTests(fit.cont))</code></pre>
<pre><code>          c0    c1    c2    c3
Down    2609  1572  1526   149
NotSig 12044 11098 13140 13729
Up       651  2634   638  1426</code></pre>
<pre class="r"><code>treat &lt;- treat(fit.cont,lfc=0.5)

dt &lt;- decideTests(treat)
summary(dt)</code></pre>
<pre><code>          c0    c1    c2    c3
Down     110   105    63     0
NotSig 15155 15083 15186 15092
Up        39   116    55   212</code></pre>
<pre class="r"><code>par(mfrow=c(2,2))
for(i in 1:ncol(mycont)){
  plotMD(treat,coef=i,status = dt[,i],hl.cex=0.5)
  abline(h=0,col=colours()[c(226)])
  lines(lowess(treat$Amean,treat$coefficients[,i]),lwd=1.5,col=4)
}</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-39-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="write-out-marker-genes-for-each-cluster" class="section level2">
<h2>Write out marker genes for each cluster</h2>
<pre class="r"><code>contnames &lt;- colnames(mycont)

for(i in 1:length(contnames)){
  topsig &lt;- topTreat(treat,coef=i,n=Inf)
  write.csv(topsig,file=paste(&quot;./output/MarkerAnalysis/Epicardial/Development/DE/Cluster-&quot;,contnames[i],&quot;.csv&quot;,sep=&quot;&quot;))
}</code></pre>
<pre class="r"><code>fdr &lt;- apply(treat$p.value, 2, function(x) p.adjust(x, method=&quot;BH&quot;))
output &lt;- data.frame(treat$genes,LogFC=treat$coefficients,AveExp=treat$Amean,tstat=treat$t, pvalue=treat$p.value, fdr=fdr)
write.csv(output,file=&quot;./output/MarkerAnalysis/Epicardial/Development/DE/MarkerAnalysis.csv&quot;)</code></pre>
</div>
</div>
<div id="perform-gene-set-testing-on-c2-and-go-sets" class="section level1">
<h1>Perform gene set testing on C2 and GO sets</h1>
<pre class="r"><code>contnames &lt;- colnames(mycont)

load(&quot;./output/RDataObjects/human_c2_v5p2.rdata&quot;)
load(&quot;./output/RDataObjects/human_c5_v5p2.rdata&quot;)

c2.id &lt;- ids2indices(Hs.c2,treat$genes$ENTREZID)
c5.id &lt;- ids2indices(Hs.c5,treat$genes$ENTREZID)

reactome.id &lt;-c2.id[grep(&quot;REACTOME&quot;,names(c2.id))]

c2.c0 &lt;- cameraPR(treat$t[,1],c2.id)
reactome.c0 &lt;- cameraPR(treat$t[,1],reactome.id)
go.c0 &lt;- cameraPR(treat$t[,1],c5.id)

for(i in 1:length(contnames)){
  write.csv(cameraPR(treat$t[,i],c2.id),file=paste(&quot;./output/MarkerAnalysis/Epicardial/Development/GeneSetTests/c2-&quot;,contnames[i],&quot;.csv&quot;,sep=&quot;&quot;))
  write.csv(cameraPR(treat$t[,i],reactome.id),file=paste(&quot;./output/MarkerAnalysis/Epicardial/Development/GeneSetTests/reactome-&quot;,contnames[i],&quot;.csv&quot;,sep=&quot;&quot;))
  write.csv(cameraPR(treat$t[,i],c5.id),file=paste(&quot;./output/MarkerAnalysis/Epicardial/Development/GeneSetTests/go-&quot;,contnames[i],&quot;.csv&quot;,sep=&quot;&quot;))
}</code></pre>
</div>
<div id="check-quality-of-clusters" class="section level1">
<h1>Check quality of clusters</h1>
<p>The quality of the clusters look good.</p>
<pre class="r"><code>par(mfrow=c(1,1))
numgenes &lt;- colSums(all.keep!=0)
boxplot(numgenes~grp)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-43-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="heatmap-of-marker-genes" class="section level1">
<h1>Heatmap of marker genes</h1>
<pre class="r"><code>sam &lt;- factor(epic.integrated$biorep,levels=c(&quot;f1&quot;,&quot;f2&quot;,&quot;f3&quot;,&quot;y1&quot;,&quot;y2&quot;,&quot;y3&quot;,&quot;a1&quot;,&quot;a2&quot;,&quot;a3&quot;))
newgrp &lt;- paste(grp,sam,sep=&quot;.&quot;)
newgrp &lt;- factor(newgrp,levels=paste(rep(levels(grp),each=9),levels(sam),sep=&quot;.&quot;))
o &lt;-order(newgrp)
clust &lt;- rep(levels(grp),each=9)
samps &lt;- rep(levels(sam),length(levels(grp)))</code></pre>
<div id="summarise-expression-across-cells" class="section level2">
<h2>Summarise expression across cells</h2>
<pre class="r"><code>sumexpr &lt;- matrix(NA,nrow=nrow(logcounts),ncol=length(levels(newgrp)))
rownames(sumexpr) &lt;- rownames(logcounts)
colnames(sumexpr) &lt;- levels(newgrp)

for(i in 1:nrow(sumexpr)){
  sumexpr[i,] &lt;- tapply(logcounts[i,],newgrp,mean)
}</code></pre>
<pre class="r"><code>sig.genes &lt;- gene.label &lt;- vector(&quot;list&quot;, length(levels(grp)))
for(i in 1:length(sig.genes)){
  top &lt;- topTreat(treat,coef=i,n=Inf)
  sig.genes[[i]] &lt;- rownames(top)[top$logFC&gt;0][1:10]
  gene.label[[i]] &lt;- paste(rownames(top)[top$logFC&gt;0][1:10],levels(grp)[i],sep=&quot;-&quot;)
} 

csig &lt;- unlist(sig.genes)
genes &lt;- unlist(gene.label)

myColors &lt;- list(Clust=NA,Sample=NA)
myColors$Clust&lt;-sample(ggplotColors(length(levels(grp))),length(levels(grp)))
names(myColors$Clust)&lt;-levels(grp)
myColors$Sample &lt;- sample(ggplotColors(length(levels(sam))),length(levels(sam)))
names(myColors$Sample) &lt;- levels(sam)


pdf(file=&quot;./output/Figures/NormalDev/epic-heatmap-siggenes-summarised-FYA-filtered.pdf&quot;,width=20,height=20,onefile = FALSE)
aheatmap(sumexpr[csig,],Rowv = NA,Colv = NA, labRow = genes,
         annCol=list(Clust=clust,Sample=samps),
         annColors=myColors,
         fontsize=16,color=&quot;-RdYlBu&quot;,
         scale=&quot;none&quot;)
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>aheatmap(sumexpr[csig,],Rowv = NA,Colv = NA, labRow = genes,
         annCol=list(Clust=clust,Sample=samps),
         annColors=myColors,
         fontsize=16,color=&quot;-RdYlBu&quot;,
         scale=&quot;none&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-46-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="heatmap-of-pre-identified-heart-genes" class="section level1">
<h1>Heatmap of pre-identified heart genes</h1>
<pre class="r"><code>hm &lt;- read.delim(&quot;./data/heart-markers-long.txt&quot;,stringsAsFactors = FALSE)
hgene &lt;- toupper(hm$Gene)
hgene &lt;- unique(hgene)

m &lt;- match(hgene,rownames(sumexpr))
m &lt;- m[!is.na(m)]

mycelltypes &lt;- hm$Celltype[match(rownames(sumexpr)[m],toupper(hm$Gene))]
mycelltypes &lt;- factor(mycelltypes)

mygenes &lt;- rownames(sumexpr)[m]
mygenelab &lt;- paste(mygenes,mycelltypes,sep=&quot;_&quot;)

pdf(file=&quot;./output/Figures/NormalDev/epic-heatmap-hmarkers-summarised-FYA-filtered.pdf&quot;,width=20,height=15,onefile = FALSE)
aheatmap(sumexpr[m,],Rowv = NA,Colv = NA, labRow = mygenelab,
         annCol=list(Clust=clust,Sample=samps),
 #        annRow=list(Celltypes=mycelltypes),
         annColors=myColors,
         fontsize=14,color=&quot;-RdYlBu&quot;)
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>aheatmap(sumexpr[m,],Rowv = NA,Colv = NA, labRow = mygenelab,
         annCol=list(Clust=clust,Sample=samps),
 #        annRow=list(Celltypes=mycelltypes),
         annColors=myColors,
         fontsize=14,color=&quot;-RdYlBu&quot;)</code></pre>
<p><img src="figure/08-ClustEpicardial.Rmd/unnamed-chunk-48-1.png" width="960" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.0 (2019-04-26)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS release 6.7 (Final)

Matrix products: default
BLAS:   /usr/local/installed/R/3.6.0/lib64/R/lib/libRblas.so
LAPACK: /usr/local/installed/R/3.6.0/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] splines   parallel  stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] dplyr_0.8.3                 clustree_0.4.0             
 [3] ggraph_1.0.2                workflowr_1.3.0            
 [5] NMF_0.21.0                  bigmemory_4.5.33           
 [7] cluster_2.1.0               rngtools_1.4               
 [9] pkgmaker_0.27               registry_0.5-1             
[11] scran_1.12.0                SingleCellExperiment_1.6.0 
[13] SummarizedExperiment_1.14.1 GenomicRanges_1.36.0       
[15] GenomeInfoDb_1.20.0         DelayedArray_0.10.0        
[17] BiocParallel_1.18.1         matrixStats_0.55.0         
[19] cowplot_1.0.0               monocle_2.12.0             
[21] DDRTree_0.1.5               irlba_2.3.3                
[23] VGAM_1.1-1                  ggplot2_3.2.1              
[25] Matrix_1.2-17               Seurat_3.0.3.9019          
[27] org.Hs.eg.db_3.8.2          AnnotationDbi_1.46.1       
[29] IRanges_2.18.1              S4Vectors_0.22.0           
[31] Biobase_2.44.0              BiocGenerics_0.30.0        
[33] RColorBrewer_1.1-2          edgeR_3.26.3               
[35] limma_3.40.2               

loaded via a namespace (and not attached):
  [1] reticulate_1.13          R.utils_2.9.0           
  [3] tidyselect_0.2.5         RSQLite_2.1.2           
  [5] htmlwidgets_1.5          grid_3.6.0              
  [7] combinat_0.0-8           docopt_0.6.1            
  [9] Rtsne_0.15               munsell_0.5.0           
 [11] codetools_0.2-16         ica_1.0-2               
 [13] statmod_1.4.30           future_1.14.0           
 [15] withr_2.1.2              colorspace_1.4-1        
 [17] fastICA_1.2-2            knitr_1.25              
 [19] ROCR_1.0-7               gbRd_0.4-11             
 [21] listenv_0.7.0            labeling_0.3            
 [23] Rdpack_0.11-0            git2r_0.26.1            
 [25] slam_0.1-45              GenomeInfoDbData_1.2.1  
 [27] polyclip_1.10-0          farver_1.1.0            
 [29] bit64_0.9-7              pheatmap_1.0.12         
 [31] rprojroot_1.3-2          vctrs_0.2.0             
 [33] xfun_0.10                R6_2.4.0                
 [35] doParallel_1.0.15        ggbeeswarm_0.6.0        
 [37] rsvd_1.0.2               locfit_1.5-9.1          
 [39] bitops_1.0-6             assertthat_0.2.1        
 [41] SDMTools_1.1-221.1       scales_1.0.0            
 [43] beeswarm_0.2.3           gtable_0.3.0            
 [45] npsurv_0.4-0             globals_0.12.4          
 [47] tidygraph_1.1.2          rlang_0.4.0             
 [49] zeallot_0.1.0            lazyeval_0.2.2          
 [51] checkmate_1.9.4          yaml_2.2.0              
 [53] reshape2_1.4.3           backports_1.1.5         
 [55] tools_3.6.0              gridBase_0.4-7          
 [57] gplots_3.0.1.1           dynamicTreeCut_1.63-1   
 [59] ggridges_0.5.1           Rcpp_1.0.2              
 [61] plyr_1.8.4               zlibbioc_1.30.0         
 [63] purrr_0.3.2              RCurl_1.95-4.12         
 [65] densityClust_0.3         pbapply_1.4-1           
 [67] viridis_0.5.1            zoo_1.8-6               
 [69] ggrepel_0.8.1            fs_1.3.1                
 [71] magrittr_1.5             data.table_1.12.4       
 [73] lmtest_0.9-37            RANN_2.6.1              
 [75] whisker_0.3-2            fitdistrplus_1.0-14     
 [77] lsei_1.2-0               evaluate_0.14           
 [79] xtable_1.8-4             sparsesvd_0.1-4         
 [81] gridExtra_2.3            HSMMSingleCell_1.4.0    
 [83] compiler_3.6.0           scater_1.12.2           
 [85] tibble_2.1.3             KernSmooth_2.23-15      
 [87] crayon_1.3.4             R.oo_1.22.0             
 [89] htmltools_0.4.0          tidyr_0.8.3             
 [91] DBI_1.0.0                tweenr_1.0.1            
 [93] MASS_7.3-51.4            R.methodsS3_1.7.1       
 [95] gdata_2.18.0             metap_1.1               
 [97] igraph_1.2.4.1           pkgconfig_2.0.3         
 [99] bigmemory.sri_0.1.3      plotly_4.9.0            
[101] foreach_1.4.7            vipor_0.4.5             
[103] dqrng_0.2.1              XVector_0.24.0          
[105] bibtex_0.4.2             stringr_1.4.0           
[107] digest_0.6.21            sctransform_0.2.0       
[109] RcppAnnoy_0.0.12         tsne_0.1-3              
[111] rmarkdown_1.14           DelayedMatrixStats_1.6.0
[113] gtools_3.8.1             nlme_3.1-141            
[115] jsonlite_1.6             BiocNeighbors_1.2.0     
[117] viridisLite_0.3.0        pillar_1.4.2            
[119] lattice_0.20-38          httr_1.4.1              
[121] survival_2.44-1.1        glue_1.3.1              
[123] qlcMatrix_0.9.7          FNN_1.1.3               
[125] png_0.1-7                iterators_1.0.12        
[127] bit_1.1-14               ggforce_0.3.0           
[129] stringi_1.4.3            blob_1.2.0              
[131] BiocSingular_1.0.0       caTools_1.17.1.2        
[133] memoise_1.1.0            future.apply_1.3.0      
[135] ape_5.3                 </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
